src/emailmessagelistmodel.cpp

```diff
@@ -335,9 +343,19 @@ void EmailMessageListModel::setSearch(const QString search)
         QMailMessageKey recipientsKey = QMailMessageKey::recipients(search, QMailDataComparator::Includes);
         QMailMessageKey fromKey = QMailMessageKey::sender(search, QMailDataComparator::Includes);
         QMailMessageKey previewKey = QMailMessageKey::preview(search, QMailDataComparator::Includes);
-        setKey(m_key & (subjectKey | recipientsKey | fromKey | previewKey));
+        m_searchKey = QMailMessageKey(m_key & (subjectKey | recipientsKey | fromKey | previewKey));
+        setKey(m_searchKey);
+
+        m_search = search;
+        // We have model filtering already via searchKey, so we pass just the current model key plus body search,
+        // otherwise results will be merged and just entries with both, fields and body matches will be returned.
+        EmailAgent::instance()->searchMessages(m_key, search, QMailSearchAction::Local, m_searchLimit);
```

Comment:
m_searchRemainingOnRemote should get cleared here, I think (and also on cancel?)